pyfr/plugins/tavg.py

```diff
@@ -338,4 +346,157 @@ def __call__(self, intg):
 
                     self.tstart_acc = intg.tcurr
 
-                self.tout_last = intg.tcurr
\ No newline at end of file
+                self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(CLIBasePlugin):
+    name = 'tavg'
+    systems = ['*']
+    formulations = ['dual', 'std']
+    cli = [
+        ('merge', 'merge_cli', ('solns', {'nargs': '*', 'help': 'Averages to merge'}), 
+                               ('output', {'help': 'Output file name'})),
+    ]
+
+    def merge_cli(self, args):
+        self.args = args
+
+        # Build up which files are needed and if std can be created
+        active = self._merge_active(args.solns)
+        merged_times = [(r.start, r.end) for r, f, s in active]
+        avg_time_inv = 1/sum([t.end - t.start for t in merged_times])
+        window = merge_intervals(merged_times)
+
+        self.std_all = std_all = all([s == 'all' for r, f, s in active])
+
+        # Merge the Data
+        for i, (rng, filename, std_mode) in enumerate(active):
+            file = NativeReader(filename)
+            dt = rng.len
+
+            if i == 0:
+                self._init_tavg_merge(file, dt)
+            else:
+                self._acc_tavg_merge(file, dt)
+
+        # Write out the result to a pyfrs file
+        out = h5py.File(args.output, 'w')
+        for k in self.dkeys:
+            if self.fexprs:
+                avg = np.transpose(self.avg_acc[k]*avg_time_inv, (1, 0, 2))
+                fun_avg = self._eval_fun_exprs(avg)
+                out[k] = np.transpose(np.vstack((avg, fun_avg)), (1, 0, 2))
+            else:
+                out[k] = self.avg_acc[k]*avg_time_inv
```

Comment:
See if you can refactor such that fexprs means doing some extra to avoid the `self.avg_acc[k]*avg_time_inv` duplication.